### PR TITLE
fix Issue 19292 - Mixin expressions should take an argument list the …

### PR DESCRIPTION
**Pull request by WalterBright**
…same as pragma(msg) does

If this works out, I'll do Mixin Statements and Mixin Declarations.

### CODE REVIEW

**Comment by thewilsonator** [comment]
Yes please! Although it won't fix [issue18472](https://issues.dlang.org/show_bug.cgi?id=18472), this should allow what it was indeed to do.

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -5282,14 +5283,38 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
...
+        {
+            foreach (ex; *exp.exps)
+            {
+                sc = sc.startCTFE();
+                auto e = ex.expressionSemantic(sc);
+                e = resolveProperties(sc, e);
+                sc = sc.endCTFE();
+
+                // allowed to contain types as well as expressions
+                e = ctfeInterpretForPragmaMsg(e);
```
Comment (thewilsonator) [inline src/dmd/expressionsem.d]: This should be tested. It also makes it clear as to what you mean by "like `pragma(msg,..);`"

**Comment by WalterBright** [comment]
Yes, and it just makes no sense that `pragma(msg, args)` and `mixin(args)` shouldn't treat `args` the same.

File: src/dmd/expressionsem.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/expressionsem.d]: As a later refactor, I intend to have `pragma(msg)` and `mixin(args)` share the same implementation, which will minimize needed tests and ensure they behave the same.

**Comment by thewilsonator** [comment]
CIs are all green. This will need a changeling entry and spec update, but they can wait until you do mixin declarations and statements and do them all at once.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**